commons/src/main/java/org/infinispan/commons/configuration/attributes/AttributeSet.java

```diff
@@ -269,6 +269,25 @@ public boolean equals(Object obj) {
       return true;
    }
 
+   public boolean matches(AttributeSet other) {
+      if (other.attributes.size() != attributes.size())
+         return false;
+      Iterator<Map.Entry<String, Attribute<?>>> i = attributes.entrySet().iterator();
+      while (i.hasNext()) {
+         Map.Entry<String, Attribute<?>> e = i.next();
+         String key = e.getKey();
+         Attribute<?> value = e.getValue();
+         if (value == null) {
+            if (!(other.attributes.get(key)==null && other.attributes.containsKey(key)))
+               return false;
+         } else {
+            if (!value.matches(other.attributes.get(key)))
```

Comment:
Hrmm should we check for null from `other.attributed.get(key)` ? It seems a lot of the `matches` implementations would throw `NullPointerException` if this happened.